src/shogun/machine/BaggingMachine.cpp

```diff
@@ -44,17 +47,53 @@ CBaggingMachine::~CBaggingMachine()
 
 CBinaryLabels* CBaggingMachine::apply_binary(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> output = apply_outputs_without_combination(data);
+
+	CMeanRule* mean_rule = new CMeanRule();
+
+	SGVector<float64_t> labels = m_combination_rule->combine(output);
+	SGVector<float64_t> probabilities = mean_rule->combine(output);
+
+	float64_t threshold = -0.5;
+	CBinaryLabels* pred = new CBinaryLabels(labels, threshold);
 
-	CBinaryLabels* pred = new CBinaryLabels(combined_vector);
 	return pred;
 }
 
 CMulticlassLabels* CBaggingMachine::apply_multiclass(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> bagged_outputs = apply_outputs_without_combination(data);
+
+	int32_t num_samples = bagged_outputs.size() / m_num_bags;
+	int32_t num_classes = dynamic_cast<CMulticlassLabels*>(m_labels)->get_num_classes();
+
+	CMulticlassLabels* pred = new CMulticlassLabels(num_samples);
```

Comment:
check whether this method is supposed to increase the ref counter for the returned object or if the caller is supposed to do that?